app/models/grda_warehouse/warehouse_reports/hud_lot.rb

```diff
@@ -0,0 +1,326 @@
+        lit_dates = literally_homeless_services.distinct.order(date: :asc).pluck(:date).map do |d|
+          [
+            d,
+            shelter_stay,
+          ]
+        end.to_h
+        extra_days = {}
+        # Fill in any gaps of < 7 days
+        lit_dates.to_a.each_with_index do |(date, _), i|
+          (next_date, _) = lit_dates.to_a[i + 1]
+          next if next_date.blank? || next_date > filter.end
+
+          if next_date < date + 7.days
+            (date..next_date).each do |d|
+              extra_days[d] = shelter_stay unless lit_dates.key?(d)
+            end
+          end
+        end
+
+        lit_dates.merge(extra_days).sort_by{|k,_| k}.to_h
+      end
+    end
+
+    # A break is added any time a PH/TH stay is started between two street stays
+    # and lasts more than 7 days
+    # 1. Occurs between two Documented street/shelter dates 7 or more days apart
+    # 2. AND does not include a Documented street/shelter date
+    # 3. AND includes 7 or more consecutive days in permanent or transitional housing.
+
+    private def breaks(un_processed_dates)
+      breaks = {}
+      a_dates = un_processed_dates.to_a
+      dates_present = a_dates.select{|_, v| v.present?}
+      dates_present.each_with_index do |(date, type), i|
+        next if shelter_stay?(type)
+        previous_i = i - 1
+        next unless shelter_stay?(dates_present[previous_i].last)
+        break unless dates_present.map(&:last)[i..].include?(shelter_stay)
+        next if next_7_days_includes?(shelter_stay, date: date, check_dates: un_processed_dates)
+        # Since we already skipped if we had a shelter stay, just make sure we have
+        # Something in all of the next seven days
+        next unless next_7_days(date: date, check_dates: un_processed_dates).count(&:present?) == 8
+
+        breaks[date] = break_marker
+      end
+      breaks
+    end
+
+    # For any date not already assigned a status, or which has a status of Self-reported street/shelter, assign a status of Self- reported/potential break for the date immediately prior to Project Start Date for any continuum enrollment where all of the following are true:
+      # Project Start Date minus 1 day is between any two dates with a status of Documented street/shelter or Self- reported street/shelter.
+      # AND Living Situation is one of the following:
+        # a. Institutional stay of more than 90 days
+        # b. OR Transitional or permanent housing situation of 7 nights or more
+
+    private def self_reported_breaks(un_processed_dates)
+      self_reported_breaks = {}
+      entries.find_each do |en|
+        project = en.project
+        date_prior_to_entry = en.first_date_in_program - 1.days
+        next unless project.coc_funded?
+        # next unless institutional_stay_longer_than_90_days?(en) || transitional_or_permanent_longer_than_7_days?(en)
+        next unless un_processed_dates[date_prior_to_entry]&.include?('street/shelter')
```

Comment:
`shelter_stay`?

Reply:
This is different, it can be a self-reported, or documented shelter stay.  We could make a method for it, but this follows the spec more closely.